drivers/windows/overlay/peerdb_windows.go

```diff
@@ -66,9 +72,15 @@ func (d *driver) peerAdd(nid, eid string, peerIP net.IP, peerIPMask net.IPMask,
 			return err
 		}
 
+		if system.GetOSVersion().Build == 14393 {
+			peerMu.Lock()
+		}
 		n.removeEndpointWithAddress(addr)
-
 		hnsresponse, err := hcsshim.HNSEndpointRequest("POST", "", string(configurationb))
+		if system.GetOSVersion().Build == 14393 {
```

Comment:
@pradipd Can we store this in a global variable and reuse the variable; the `GetOSVersion` will result of a system call every time it's called [GetOSVersion](https://github.com/dani-docker/libnetwork/blob/d768021dd5e94d5a14acca094153e5bf38178cfd/vendor/golang.org/x/sys/windows/zsyscall_windows.go#L562`)

something like that?
```
isWin2016 = system.GetOSVersion().Build == 14393
```